aquarius/events/events_monitor.py

```diff
@@ -574,70 +505,105 @@ def get_assets_in_chain(self):
-            all_logs.extend(logs)
-            if (_from - from_block) % 1000 == 0:
-                logger.debug(
-                    f"Searched blocks {_from} to {_to} on chain {self._chain_id}; "
-                    f"{len(all_logs)} {event_name} events detected so far."
-                )
+        except Exception as e:
+            if from_block < to_block:
+                # splitting in two might help, so rely on that
+                raise Exception(f"Failed to get events for multiple blocks")
+            else:
+                # Since there is only one block, and we failed to get all events, we need to try to take them one by one
+                # if any call fails, there is nothing more we can do  (ie:  failed to get only transfer events from block X)
+                for topic in EventTypes.hashes:
+                    filter_params = {
+                        "topics": [topic],
+                        "fromBlock": from_block,
+                        "toBlock": to_block,
+                    }
+                    try:
+                        logs = self._web3.eth.get_logs(filter_params)
+                        self.process_logs(logs, to_block)
+                    except Exception as e:
+                        logger.error(
+                            f"Failed to fetch {EventTypes.hashes[topic]['type']} logs from block {from_block}.  Bailing out.."
+                        )
+                return
+        try:
+            self.process_logs(logs, to_block)
+        except Exception as e:
+            logger.error(
+                f"Failed to process logs {from_block} to {to_block}. Error: {e}"
+            )
+        # finally, stored last block in ES
+        self.store_last_processed_block(to_block)
 
-            # Prepare for next chunk
-            _from = _to + 1
-            _to = min(_from + chunk_size - 1, to_block)
-            filter_params.update({"fromBlock": _from, "toBlock": _to})
+    def process_logs(self, logs, to_block):
+        """Process a list of logs"""
+        processor_args = [
+            self._es_instance,
+            self._web3,
+            self._allowed_publishers,
+            self.purgatory,
+            self._chain_id,
+        ]
 
-        logger.info(
-            f"Finished searching for {event_name} events on chain {self._chain_id} "
-            f"in blocks {from_block} to {to_block}. "
-            f"{len(all_logs)} {event_name} events detected."
-        )
+        logger.info(f"Processing {len(logs)} events ...")
+        for event in logs:
+            match = EventTypes.hashes.get(event.topics[0].hex(), None)
+            if match is None:
+                logger.warning(f"Unknown event ")
+                logger.warning([event])
+                continue
+            if match["type"] == EventTypes.EVENT_TRANSFER:
+                self.handle_transfer_ownership([event])
+            elif match["type"] == EventTypes.EVENT_METADATA_CREATED:
```

Comment:
@alexcos20 @calina-c 
guys, to me the "for event in logs:" in line 532 together with the nested "for item in EventTypes.list" from line 536 makes me think that this is O² and that depending on the number of logs this will overflow. Can I ask your opinion? Am I missing something? thanks!

Reply:
@calina-c  - done [here](https://github.com/oceanprotocol/aquarius/pull/928/commits/607dfb126d3d140428668c3b098fe75f74ca0e6f)